props/saber_BC_buttons.h

```diff
@@ -477,738 +1495,1272 @@ class SaberBCButtons : public PROP_INHERIT_PREFIX PropBase {
         thrust_begin_millis_ = millis();
       }
-
-    // Mute timer check - play optional mute.wav first.
-    if (mute_timer_.timerCheck()) {
-      DoMute();
+    } else {
+      thrust_begin_millis_ = millis();
+    }
+// Timers
+    // Play optional mute.wav first.
+    if (mute_all_timer_.timerCheck()) {
+      if (SetMute(true)) {
+        unmute_on_deactivation_ = true;
+        muted_ = true;
+        TurnOnHelper();
+      }
+    }
+#ifdef BC_DUAL_BLADES
+    if (mute_mainBlade_timer_.timerCheck()) {
+      if (SetMute(true)) {
+        unmute_on_deactivation_ = true;
+        muted_ = true;
+        TurnBladeOn(BC_MAIN_BLADE_SET);
+        PVLOG_NORMAL << "** Main Blade Turned ON Muted\n";
+      }
+    }
+    if (mute_secondBlade_timer_.timerCheck()) {
+      if (SetMute(true)) {
+        unmute_on_deactivation_ = true;
+        muted_ = true;
+        TurnBladeOn(BC_SECOND_BLADE_SET);
+        PVLOG_NORMAL << "** Second Blade Turned ON Muted\n";
+      }
     }
-    // Scroll Presets timer check - avoid beep/wav overlap
+#endif
+    // Allow beep to be heard.
     if (scroll_presets_timer_.timerCheck() && scroll_presets_) {
         SaberBase::DoEffect(EFFECT_NEWFONT, 0);
     }
+    if (twist_timer_.timerCheck() && saved_twist_) {
+      DoSavedTwist();
+    }
+    quote_delay_timer_.timerCheck();
   }  // Loop()
 
 #ifdef SPEAK_BLADE_ID
   void SpeakBladeID(float id) override {
     if (&SFX_mnum) {
       sound_library_.SayNumber(id, SAY_WHOLE);
     } else {
-      PVLOG_NORMAL << "No mnum.wav number prompts found.\n";
+      PVLOG_NORMAL << "** No mnum.wav number prompts found.\n";
       beeper.Beep(0.25, 2000.0);
     }
   }
 #endif
 
-// Revert colorchange witout saving (reset to Variation == 0)
-  void ResetColorChangeMode() {
-    if (!current_style()) return;
-      PVLOG_NORMAL << "Reset Color Variation\n";
-      SetVariation(0);
-      PVLOG_NORMAL << "Color change mode done, variation = " << SaberBase::GetCurrentVariation() << "\n";
-      SaberBase::SetColorChangeMode(SaberBase::COLOR_CHANGE_MODE_NONE);
-  }
-
-// Volume Menu
-    void VolumeMenu() {
-    if (!mode_volume_) {
-      current_menu_angle_ = fusor.angle2();
-      mode_volume_ = true;
-      if (SFX_vmbegin) {
-        sound_library_.SayEnterVolumeMenu();
-      } else {
-        beeper.Beep(0.1, 1000);
-        beeper.Beep(0.1, 2000);
-        beeper.Beep(0.1, 3000);
-      }
-      PVLOG_NORMAL << "Enter Volume Menu\n";
-      SaberBase::DoEffect(EFFECT_VOLUME_LEVEL, 0);
-    } else {
-      mode_volume_ = false;
-      if (SFX_vmend) {
-        sound_library_.SayVolumeMenuEnd();
-      } else {
-        beeper.Beep(0.1, 3000);
-        beeper.Beep(0.1, 2000);
-        beeper.Beep(0.1, 1000);
-      }
-      PVLOG_NORMAL << "Exit Volume Menu\n";
+  void DetectMenuTurn() {
+    float a = fusor.angle2() - current_menu_angle_;
+    if (isPointingUp()) return;
+    // Keep the rotational angle within range of -180 to 180.
```

Comment:
Seems confusing to use degrees in the comment and radians in the code...

Reply:
It was a note-to-self really when realizing what it was doing. 
No harm in keeping it, right? But I added "in terms of radians".